common/faults/faults.c

```diff
@@ -0,0 +1,461 @@
+/**
+ * @file faults.c
+ * @author Aditya Anand (anand89@purdue.edu)
+ * @brief Creating a library of faults to create an easy to debug atmosphere on the car
+ * @version 0.1
+ * @date 2022-05-11
+ *
+ * @copyright Copyright (c) 2022
+ *
+ */
+#include "faults.h"
+
+
+//BEGIN AUTO INCLUDES
+#if FAULT_NODE_NAME == 0
+	#include "source/main_module/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 1
+	#include "source/driveline/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 2
+	#include "source/dashboard/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 3
+	#include "source/precharge/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 4
+	#include "source/torque_vector/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 5
+	#include "source/l4_testing/can/can_parse.h"
+#endif
+//END AUTO INCLUDES
+
+//BEGIN AUTO FAULT INFO ARRAY DEFS
+int idArray[TOTAL_NUM_FAULTS] = { ID_BATT_FLOW_FAULT, ID_DRIVE_FLOW_FAULT, ID_MAIN_COMM_FAULT, ID_LV_DEAD_FAULT, ID_MOT_FRONT_OT_FAULT, ID_WLSPD_L_FAULT,
```

Comment:
Specifically for the idArray, it doesn't matter. All I really care about is the first 4 bits - unsigned int representing node #, and the last 12 bits - representing the index in the fault array. Since I don't care about the value itself, all I need is a 16 bit representation - and int works just fine for that.